wandb/sdk/internal/internal_api.py

```diff
@@ -322,10 +347,21 @@ def viewer_server_info(self):
             _CLI_QUERY_
         }
         """
-        query_new = gql(query_str.replace("_CLI_QUERY_", cli_query))
-        query_old = gql(query_str.replace("_CLI_QUERY_", ""))
-
-        for query in query_new, query_old:
+        queries = []
+        queries.append(
+            gql(
+                query_str.replace("_CLI_QUERY_", cli_query).replace(
+                    "_LOCAL_QUERY_", local_query
+                )
+            )
+        )
+        queries.append(
+            gql(
+                query_str.replace("_CLI_QUERY_", cli_query).replace("_LOCAL_QUERY_", "")
+            )
+        )
+        queries.append(gql(query_str.replace("_CLI_QUERY_", "")))
+        for query in queries:
```

Comment:
it would be nice to not have to do these queries in serial like this. what if this function itself does the introspect and dynamically decides whether to include _LOCAL_QUERY_? Then the caller can just check if all the expected fields are present.

Reply:
Yeah... I was thinking of doing this... but was worried that it might break things (so did super conservative change), but I can do it.

Reply:
Seems like the these changes break windows run... so i'm reverting them for now

Reply:
Agree with Adrian, if we are using schema introspection this query can be resolved in 2 roundtrips vs 2-4, because the single introspection request should have enough info to determine which of these is safe to make.

It looks like you need two pieces of info to construct this query:

1. does `Query.serverInfo` exist?
2. does `ServerInfo.latestLocalVersionInfo` exist?

You can query for both of these in a single introspection request using aliases:

```
query ProbeServerCapabilities {
  QueryType: __type(name: "Query") { ...fieldData }
  ServerInfoType: __type(name: "ServerInfo") { ...fieldData }
}

fragment fieldData on __Type {
  fields {
    name
  }
}
```

And your response would come back as:

```
{
  QueryType: {
    fields: [ { name: 'x' }, { name: 'y' }, ... ],
  }
  ServerInfoType: {
    fields: [ { name: 'x' }, { name: 'y' }, ... ],
  }
}
```

Reply:
Yeah i tried to do something similar, but windows test were failing (the only ones), so i reverted.
I can try to do it again, but it was really hard to debug the windows failure.